### DISCUSSION TITLE
How can we store ldap group membership in Zitadel?

### DISCUSSION THREAD

**Discussion by nikzen** [zitadel/zitadel#5882] [General]
Hey,

we are using a service to provision ldap users to Zitadel. We now want to move one step further and also sync ldap group memberships into Zitadel. Do you have an idea how to do it?

**Comment by fforootd** [comment]
Sorry this slipped through my eyes somehow.

You have two options there:

1. Use ZITADEL Roles and User Grants to map your Groups to the users (a group = role, and membership = user grant)
2. Store the data in the metadata section of a user (a key value storage that can hold arbitrary data but is not yet searchable)
